packages/bot-web-ui/src/components/toolbox/toolbox.jsx

```diff
@@ -0,0 +1,192 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Field as FormField, Formik, Form } from 'formik';
+import { Drawer, Input, Icon, Text } from '@deriv/components';
+import { localize } from '@deriv/translations';
+import { ToolboxItems } from './toolbox-items.jsx';
+import { connect } from '../../stores/connect';
+import { popover_zindex } from '../../constants/z-indexes';
+
+const SearchBox = ({ is_search_loading, onSearch, onSearchBlur, onSearchClear, onSearchKeyUp }) => (
+    <div className='db-toolbox__search'>
+        <Formik initialValues={{ search: '' }} onSubmit={onSearch}>
+            {({ submitForm, values: { search }, setFieldValue }) => (
+                <Form>
+                    <FormField name='search'>
+                        {({ field }) => (
+                            <Input
+                                {...field}
+                                className='db-toolbox__search-field'
+                                type='text'
+                                name='search'
+                                placeholder={localize('Search')}
+                                onKeyUp={() => onSearchKeyUp(submitForm)}
+                                onFocus={submitForm}
+                                onBlur={onSearchBlur}
+                                leading_icon={
+                                    (search &&
+                                        (is_search_loading ? (
+                                            <div className='loader' />
+                                        ) : (
+                                            <Icon
+                                                icon='IcCloseCircle'
+                                                onClick={() => onSearchClear(setFieldValue)}
+                                                color='secondary'
+                                            />
+                                        ))) ||
+                                    (!search && <Icon icon='IcSearch' />)
+                                }
+                            />
+                        )}
+                    </FormField>
+                </Form>
+            )}
+        </Formik>
+    </div>
+);
+
+const Toolbox = ({
+    onMount,
+    onUnmount,
+    hasSubCategory,
+    is_mobile,
+    is_search_loading,
+    is_toolbox_open,
+    onSearch,
+    onSearchBlur,
+    onSearchClear,
+    onSearchKeyUp,
+    onToolboxItemClick,
+    onToolboxItemExpand,
+    sub_category_index,
+    toggleDrawer,
+    toolbox_dom,
+}) => {
+    const toolbox_ref = React.useRef(ToolboxItems);
+
+    React.useEffect(() => {
+        onMount(toolbox_ref);
+        return () => onUnmount();
+    }, [onMount, onUnmount]);
+
```

Comment:
```JS
if (is_mobile) {
    return null;
}
```

Reply:
I've put a check in `toolbox-store.js` inside `onMount` since we need it to adjust the workspace.

Reply:
The effect will still run if you return `null`.

Reply:
Got it. This is done.